pkgs/applications/audio/easyeffects/default.nix

```diff
@@ -49,15 +52,18 @@ stdenv.mkDerivation rec {
     ninja
     pkg-config
     python3
+    tbb
```

Comment:
How did you determine it goes to `nativeBuildInputs`?

Reply:
Good catch - when I made this change, I was under impression that native inputs meant "build time" dependencies. But after thinking about it in the cross-compilation context (which I had learned later) I am not sure what should go where anymore. Is there an algorithm to accurately decide this?

Reply:
That is about right `nativeBuildInputs` is about dependencies used, or rather, executed at build time. So mostly programs used for locating dependencies, build tools, linkers, documentation generators… `tbb` is a library and while it is possible to build and run an executable linking a library at build time, it is quite rare. So unless you have a reason to suspect a library to be used this way (e.g. `libxslt` contains `xsltproc` program often used for building man pages from DocBook files), it is most likely a runtime dependency belonging to `buildInputs`.

See also https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/build-tools-in-build-inputs.md

Reply:
Is it safe to always put any new dependency in buildInputs first? And only when it breaks move it to nativeBuildInputs?

Reply:
> so it should not be left in the code

Does `it` reference to the `strictDeps` call? I am roughly understanding what you wrote as "Use `strictDeps` in testing, but make sure it is removed in the final code."